lib/mayaUsd/render/vp2RenderDelegate/mayaPrimCommon.cpp

```diff
@@ -564,14 +572,43 @@ void MayaUsdRPrim::_SyncSharedData(
     }
 
     if (HdChangeTracker::IsVisibilityDirty(*dirtyBits, id)) {
-        sharedData.visible = delegate->GetVisible(id);
+        bool usdVisibility = delegate->GetVisible(id);
 
         // Invisible rprims don't get calls to Sync or _PropagateDirtyBits while
         // they are invisible. This means that when a prim goes from visible to
         // invisible that we must update every repr, because if we switch reprs while
         // invisible we'll get no chance to update!
-        if (!sharedData.visible)
+        if (!usdVisibility)
             _MakeOtherReprRenderItemsInvisible(reprToken, reprs);
+
+        bool displayLayerVisibility = true; //objects in the default display layer are visible
+#ifdef MAYA_HAS_DISPLAY_LAYER_API
+        // Maya Display Layers do not have a representation in USD, so a prim can be
+        // visible from USD's point of view, but hidden from Maya's point of view.
+        // In Maya Display Layer visibility really hides the render items vs. using
+        // render item filtering (like for example the "Show" menu). So we want to
+        // set the "enabled" state of the MRenderItem.
+
+        // Get all the display layers the object is affected by. If any of those layers
+        // are invisible, the object is invisible.
+        MFnDisplayLayerManager displayLayerManager(MFnDisplayLayerManager::currentDisplayLayerManager());
+        MStatus                status;
+        auto* const            param = static_cast<HdVP2RenderParam*>(_delegate->GetRenderParam());
+        ProxyRenderDelegate&   drawScene = param->GetDrawScene();
+        MDagPath               proxyDagPath = drawScene.GetProxyShapeDagPath();
+        MString pathString = Ufe::PathString::string(MayaUsd::ufe::dagPathToUfe(proxyDagPath)).c_str();
+        pathString += Ufe::PathString::pathSegmentSeparator().c_str();
+        pathString += _PrimSegmentString[0];
+        MObjectArray ancestorDisplayLayers
+            = displayLayerManager.getAncestorLayersInclusive(pathString, &status);
+        for(unsigned int i=0; i<ancestorDisplayLayers.length() && displayLayerVisibility; i++) {
+            MFnDependencyNode displayLayerNodeFn(ancestorDisplayLayers[i]);
+            MPlug layerEnabled = displayLayerNodeFn.findPlug("enabled");
+            MPlug layerVisible = displayLayerNodeFn.findPlug("visibility");
+            displayLayerVisibility &= layerEnabled.asBool() ? layerVisible.asBool() : true;
+        }
```

Comment:
Well have to use a similar pattern to find display layers in other places in the code to get other display layer features working. We may decide that it is worthwhile grabbing all the layers once and holding them until the end of the sync call.